app/controllers/cart_controller.rb

```diff
@@ -0,0 +1,68 @@
+class CartController < ApplicationController
+  def index
+    @carts = session[:cart].to_a
+    count_item
+    total_sum
+  end
+
+  def create
+    session[:cart] ||= {}
+    cart = session[:cart]
+    product_id = params[:product_id].to_s
+    if cart[product_id]
+      cart[product_id] += 1
+    else
+      cart[product_id] = 1
+    end
+    respond_to do |format|
+      format.html{redirect_to cart_index_path}
+      format.js
+    end
+  end
+
+  def update
+    @product_id = params[:id]
+    check_update
+  end
+
+  def remove_cart_item
+    product_id = params[:id]
+    session[:cart].delete product_id
+    flash[:danger] = t "message.cart_deleted"
+    redirect_to cart_index_path
+  end
+
+  def remove_all_item
+    session[:cart].clear
+    flash[:danger] = t "message.cart_deleted"
+    redirect_to cart_index_path
+  end
+
+  private
+
+  def check_update
+    if params[:qty].to_i >= 1 && params[:qty].to_i <= 10
+      session[:cart][@product_id] = params[:qty].to_i
+      flash[:success] = t "message.cart_update_success"
+    else
+      flash[:warning] = t "message.cart_update_fail"
+    end
+    redirect_to cart_index_path
+  end
+
+  def count_item
+    @count = 0
+    @carts.each do |(_pro_id, qty)|
+      @count += qty
+    end
+    @count
+  end
+
+  def total_sum
+    @sum_total = 0
+    @carts.each do |(pro_id, _qty)|
+      @sum_total += (Product.find_by(id: pro_id).price * @count)
```

Comment:
Sao không where IN 1 lần từ đầu luôn mà for each cái @carts chọt vào DB để query product liên tục thế này e?

Reply:
Find by mà nil có .price đc ko chú

Reply:
dạ em sửa lại rồi ạ. Cảm ơ anh chị